tracepoint_decode/src/display.rs

```diff
@@ -0,0 +1,572 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//! `fmt::Display` helpers for using  tracepoint_decode types with format
+//! macros like [`write!`] and [`format_args!`].
+
+use core::fmt;
+use core::fmt::Write;
+
+use eventheader_types::*;
+
+use crate::charconv;
+use crate::enumerator;
+use crate::filters;
+use crate::filters::Filter;
+use crate::perf_abi;
+use crate::perf_event_desc;
+use crate::perf_session;
+use crate::writers;
+
+use crate::PerfConvertOptions;
+use crate::PerfMetaOptions;
+
+/// Display implementation that JSON-escapes the provided input string.
+/// This escapes control chars, quotes, and backslashes. For example,
+/// the string `Hello, "world"!` would be displayed as `Hello, \"world\"!`.
+pub struct JsonEscapeDisplay<'str> {
```

Comment:
Note: New type added to make it easy for the user to JSON-escape a string. Basically acts like [std::char::EscapeUnicode](https://doc.rust-lang.org/std/char/struct.EscapeUnicode.html) but it uses JSON escape rules instead of C escape rules.